File: example_usage.py
#!/usr/bin/env python3
"""
أمثلة لاستخدام أداة تصدير معايير التصنيف
"""

import asyncio
from table_pdf_exporter_advanced import AdvancedTablePDFExporter


async def example_1_basic_usage():
    """المثال 1️⃣: الاستخدام الأساسي"""
    print('\n╔════════════════════════════════════════╗')
    print('║       المثال 1️⃣: الاستخدام الأساسي       ║')
    print('╚════════════════════════════════════════╝\n')

    decisions = [
        {'name': 'قرار_21', 'decisionNumber': '21'},
    ]

    exporter = AdvancedTablePDFExporter('admin', 'password123')

    try:
        results = await exporter.run(decisions)
        for r in results:
            if r['status'] == 'success':
                print(f"✅ {r['siteName']}: {r['pdfPath']}")
            else:
                print(f"❌ {r['error']}")
    except Exception as e:
        print(f'❌ خطأ: {e}')


async def example_2_multiple_decisions():
    """المثال 2️⃣: معالجة عدة قرارات"""
    print('\n╔════════════════════════════════════════╗')
    print('║      المثال 2️⃣: عدة قرارات في دفعة        ║')
    print('╚════════════════════════════════════════╝\n')

    decisions = [
        {'name': 'قرار_1', 'decisionNumber': '21'},
        {'name': 'قرار_2', 'decisionNumber': '22'},
        {'name': 'قرار_3', 'decisionNumber': '23'},
    ]

    exporter = AdvancedTablePDFExporter('admin', 'password123')

    try:
        results = await exporter.run(decisions)

        print('\n╔════════════════════════════════════════╗')
        print('║             النتائج النهائية           ║')
        print('╚════════════════════════════════════════╝')

        for i, r in enumerate(results, 1):
            if r['status'] == 'success':
                print(f"{i}. ✅ {r['siteName']}")
                print(f"   📄 PDF: {r['pdfPath']}")
                print(f"   🖼️  PNG: {r['pngPath']}\n")
            else:
                print(f"{i}. ❌ {r['name']}: {r['error']}\n")

        success_count = sum(1 for r in results if r['status'] == 'success')
        print(f'📊 النتيجة: {success_count}/{len(results)} بنجاح')

    except Exception as e:
        print(f'❌ خطأ: {e}')


async def example_3_custom_settings():
    """المثال 3️⃣: إعدادات مخصصة"""
    print('\n╔════════════════════════════════════════╗')
    print('║      المثال 3️⃣: إعدادات مخصصة            ║')
    print('╚════════════════════════════════════════╝\n')

    decisions = [
        {'name': 'قرار_21', 'decisionNumber': '21'},
    ]

    exporter = AdvancedTablePDFExporter(
        username='admin',
        password='password123',
        options={
            'outputDir': './custom-exports',  # مجلد مخصص
            'headless': False,  # عرض المتصفح
            'navigationTimeout': 60000,  # 60 ثانية
        }
    )

    try:
        results = await exporter.run(decisions)
        for r in results:
            if r['status'] == 'success':
                print(f"✅ تم التصدير: {r['pdfPath']}")
            else:
                print(f"❌ خطأ: {r['error']}")
    except Exception as e:
        print(f'❌ خطأ: {e}')


async def example_4_manual_steps():
    """المثال 4️⃣: خطوات يدوية للتحكم الكامل"""
    print('\n╔════════════════════════════════════════╗')
    print('║      المثال 4️⃣: خطوات يدوية مفصلة         ║')
    print('╚════════════════════════════════════════╝\n')

    exporter = AdvancedTablePDFExporter('admin', 'password123')

    try:
        # 1. تهيئة
        print('1️⃣ تهيئة المتصفح...')
        await exporter.init()

        # 2. تسجيل دخول
        print('2️⃣ تسجيل الدخول...')
        await exporter.login()

        # 3. تحميل البيانات
        print('3️⃣ تحميل البيانات...')
        await exporter.scroll_table_and_load_data()

        # 4. إدخال رقم القرار
        print('4️⃣ إدخال رقم القرار...')
        await exporter.select_decision_number('21')

        # 5. البحث
        print('5️⃣ الضغط على البحث...')
        await exporter.click_search_button()

        # 6. فتح الاستمارة
        print('6️⃣ فتح الاستمارة...')
        await exporter.open_first_form()

        # 7. الانتقال لمعايير التصنيف
        print('7️⃣ الانتقال لمعايير التصنيف...')
        await exporter.navigate_to_classification_criteria()

        # 8. استخراج الجدول
        print('8️⃣ استخراج معلومات الجدول...')
        table_info = await exporter.extract_red_box_table()
        print(f'   حجم الجدول: {table_info["width"]}x{table_info["height"]}px')

        # 9. استخراج اسم الموقع
        print('9️⃣ استخراج اسم الموقع...')
        site_name = await exporter.get_site_name()
        print(f'   الموقع: {site_name}')

        # 10. تصدير PDF
        print('🔟 تصدير PDF...')
        pdf_path = await exporter.export_table_as_clean_pdf(f'{site_name}.pdf', table_info)
        print(f'    ✅ {pdf_path}')

        # 11. حفظ لقطة شاشة
        print('1️⃣1️⃣ حفظ لقطة شاشة...')
        png_path = await exporter.capture_table_screenshot(f'{site_name}.png', table_info)
        print(f'    ✅ {png_path}')

        # 12. إغلاق
        print('1️⃣2️⃣ إغلاق المتصفح...')
        await exporter.close()

        print('\n✅ انتهت جميع الخطوات بنجاح!')

    except Exception as e:
        await exporter.close()
        print(f'❌ خطأ: {e}')


async def example_5_batch_processing():
    """المثال 5️⃣: معالجة دفعية كبيرة"""
    print('\n╔════════════════════════════════════════╗')
    print('║      المثال 5️⃣: معالجة دفعية كبيرة        ║')
    print('╚════════════════════════════════════════╝\n')

    # إنشاء قائمة كبيرة من القرارات
    decisions = []
    for i in range(1, 11):  # 10 قرارات
        decisions.append({
            'name': f'قرار_{i}',
            'decisionNumber': str(20 + i),
        })

    exporter = AdvancedTablePDFExporter('admin', 'password123')

    try:
        results = await exporter.run(decisions)

        print('\n╔════════════════════════════════════════╗')
        print('║             الملخص النهائي             ║')
        print('╚════════════════════════════════════════╝')

        success_results = [r for r in results if r['status'] == 'success']
        failed_results = [r for r in results if r['status'] == 'failed']

        print(f'\n✅ نجح: {len(success_results)}/{len(results)}')
        for r in success_results:
            print(f'   • {r["siteName"]}')

        if failed_results:
            print(f'\n❌ فشل: {len(failed_results)}/{len(results)}')
            for r in failed_results:
                print(f'   • {r["name"]}: {r["error"]}')

        print(f'\n📊 إجمالي الملفات: {len(success_results) * 2}')
        print(f'   (PDF و PNG لكل استمارة)')

    except Exception as e:
        print(f'❌ خطأ: {e}')


async def main():
    """تشغيل الأمثلة"""
    import sys

    if len(sys.argv) < 2:
        print('\nأمثلة لاستخدام أداة التصدير:')
        print('  python example_usage.py 1  - استخدام أساسي')
        print('  python example_usage.py 2  - عدة قرارات')
        print('  python example_usage.py 3  - إعدادات مخصصة')
        print('  python example_usage.py 4  - خطوات يدوية')
        print('  python example_usage.py 5  - معالجة دفعية\n')
        sys.exit(1)

    example_num = sys.argv[1]

    examples = {
        '1': example_1_basic_usage,
        '2': example_2_multiple_decisions,
        '3': example_3_custom_settings,
        '4': example_4_manual_steps,
        '5': example_5_batch_processing,
    }

    if example_num in examples:
        await examples[example_num]()
    else:
        print(f'❌ المثال {example_num} غير موجود')
        print('   اختر من: 1, 2, 3, 4, 5')


if __name__ == '__main__':
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        print('\n\n⚠️ تم الإيقاف بواسطة المستخدم')

File: example-usage.js
/**
 * مثال على استخدام أداة تصدير جداول STAR1
 *
 * هذا المثال يوضح كيفية:
 * 1. تسجيل الدخول تلقائياً
 * 2. إدخال رقم قرار التسجيل
 * 3. فتح الاستمارات
 * 4. الذهاب إلى معايير التصنيف
 * 5. تصدير الجداول إلى PDF
 */

const AdvancedTablePDFExporter = require('./table-pdf-exporter-advanced');

async function example1_BasicUsage() {
  console.log('═══════════════════════════════════════════');
  console.log('  مثال 1: الاستخدام الأساسي');
  console.log('═══════════════════════════════════════════\n');

  const exporter = new AdvancedTablePDFExporter('username', 'password');

  const decisions = [
    { decisionNumber: '21', name: 'قرار_21' },
    { decisionNumber: '22', name: 'قرار_22' },
  ];

  try {
    const results = await exporter.run(decisions);

    console.log('\n✅ النتائج:');
    results.forEach(r => {
      if (r.status === 'success') {
        console.log(`  ✓ ${r.decisionNumber}: ${r.pdfPath}`);
      } else {
        console.log(`  ✗ ${r.decisionNumber}: ${r.error}`);
      }
    });
  } catch (error) {
    console.error('❌ خطأ:', error.message);
  }
}

async function example2_CustomSettings() {
  console.log('\n═══════════════════════════════════════════');
  console.log('  مثال 2: إعدادات مخصصة');
  console.log('═══════════════════════════════════════════\n');

  const exporter = new AdvancedTablePDFExporter('username', 'password', {
    baseUrl: 'https://heritage2.itqan-consultant.com/Web/App/Pools/DataEdit/19112/10',
    outputDir: './my-exports',
    headless: false, // عرض المتصفح للمراقبة
    navigationTimeout: 45000,
    maxRetries: 5,
  });

  const decisions = [
    {
      decisionNumber: '21',
      name: 'قرار_ترشيح_رئيسي',
      url: 'https://heritage2.itqan-consultant.com/Web/App/Pools/DataEdit/19112/10' // URL اختياري
    },
  ];

  try {
    const results = await exporter.run(decisions);
    console.log('✅ تم التصدير بنجاح!');
  } catch (error) {
    console.error('❌ خطأ:', error.message);
  }
}

async function example3_ManualSteps() {
  console.log('\n═══════════════════════════════════════════');
  console.log('  مثال 3: خطوات يدوية للتحكم الكامل');
  console.log('═══════════════════════════════════════════\n');

  const exporter = new AdvancedTablePDFExporter('username', 'password');

  try {
    // الخطوة 1: تهيئة المتصفح
    console.log('1. تهيئة المتصفح...');
    await exporter.init();

    // الخطوة 2: تسجيل الدخول
    console.log('2. تسجيل الدخول...');
    await exporter.login();

    // الخطوة 3: إدخال رقم القرار
    console.log('3. إدخال رقم القرار 21...');
    await exporter.selectDecisionNumber('21');

    // الخطوة 4: فتح الاستمارة الأولى
    console.log('4. فتح الاستمارة...');
    await exporter.openFirstForm();

    // الخطوة 5: الانتقال إلى معايير التصنيف
    console.log('5. الانتقال إلى معايير التصنيف...');
    await exporter.navigateToClassificationCriteria();

    // الخطوة 6: تحميل كل البيانات
    console.log('6. تحميل البيانات...');
    await exporter.scrollTableAndLoadData();

    // الخطوة 7: استخراج الجدول
    console.log('7. استخراج الجدول...');
    const redBoxInfo = await exporter.extractRedBoxTable();

    // الخطوة 8: تصدير PDF
    console.log('8. تصدير PDF...');
    const pdfPath = await exporter.exportTableAsCleanPDF('قرار_21.pdf', redBoxInfo);

    // الخطوة 9: تصدير لقطة شاشة
    console.log('9. تصدير لقطة شاشة...');
    const pngPath = await exporter.captureTableScreenshot('قرار_21.png', redBoxInfo);

    console.log(`\n✅ تم الانتهاء!`);
    console.log(`   PDF: ${pdfPath}`);
    console.log(`   PNG: ${pngPath}`);

    // الخطوة 10: إغلاق المتصفح
    console.log('10. إغلاق المتصفح...');
    await exporter.close();
  } catch (error) {
    console.error('❌ خطأ:', error.message);
    await exporter.close();
  }
}

async function example4_BatchProcessing() {
  console.log('\n═══════════════════════════════════════════');
  console.log('  مثال 4: معالجة دفعة كبيرة من القرارات');
  console.log('═══════════════════════════════════════════\n');

  const exporter = new AdvancedTablePDFExporter('username', 'password', {
    outputDir: './batch-exports',
    headless: true,
  });

  // قرارات متعددة
  const decisions = Array.from({ length: 10 }, (_, i) => ({
    decisionNumber: String(21 + i),
    name: `قرار_${21 + i}`
  }));

  try {
    console.log(`📋 جاري معالجة ${decisions.length} قرار...`);
    const results = await exporter.run(decisions);

    const successful = results.filter(r => r.status === 'success').length;
    console.log(`\n✅ النتيجة: ${successful}/${results.length}`);

    // عرض ملخص
    results.forEach((r, i) => {
      const icon = r.status === 'success' ? '✓' : '✗';
      console.log(`  ${icon} قرار ${r.decisionNumber}`);
    });
  } catch (error) {
    console.error('❌ خطأ:', error.message);
  }
}

// اختيار أي مثال تريد تشغيله
const exampleNumber = process.argv[2] || '1';

(async () => {
  switch (exampleNumber) {
    case '1':
      await example1_BasicUsage();
      break;
    case '2':
      await example2_CustomSettings();
      break;
    case '3':
      await example3_ManualSteps();
      break;
    case '4':
      await example4_BatchProcessing();
      break;
    default:
      console.log('استخدام: node example-usage.js [1|2|3|4]');
      console.log('  1 - الاستخدام الأساسي');
      console.log('  2 - إعدادات مخصصة');
      console.log('  3 - خطوات يدوية');
      console.log('  4 - معالجة دفعة');
  }
})();
